config/genesis.go

```diff
@@ -0,0 +1,145 @@
+}
+
+func defaultGenesisConfig() *GenesisConfig {
+	return &GenesisConfig{
+		Accounts:    apiConfig.DefaultGenesisAccountConfig(),
+		GenesisTime: DefaultGenesisTime(),
+		GoldenATXID: DefaultGoldenATXId().Hash32().Hex(),
+		ExtraData:   DefaultGenesisExtraData(),
+		Path:        DefaultGenesisDataDir(),
+	}
+}
+
+func DefaultTestnetGenesisConfig() *GenesisConfig {
+	return &GenesisConfig{
+		Accounts: &apiConfig.GenesisAccountConfig{
+			Data: map[string]uint64{
+				"stest1qqqqqqygdpsq62p4qxfyng8h2mm4f4d94vt7huqqu9mz3": 100000000000000000,
+				"stest1qqqqqqylzg8ypces4llx4gnat0dyntqfvr0h6mcprcz66": 100000000000000000,
+				"stest1qqqqqq90akdpc97206485eu4m0rmacd3mxfv0wsdrea6k": 100000000000000000,
+				"stest1qqqqqq9jpsarr7tnyv0qr0edddwqpg3vcya4cccauypts": 100000000000000000,
+				"stest1qqqqqq8lpq7f5ghqt569nvpl8kldv8r66ms2yzgudsd5t": 100000000000000000,
+			},
+		},
+		GenesisTime: DefaultTestnetGenesisTime(),
+		ExtraData:   DefaultTestnetGenesisExtraData(),
+		GoldenATXID: DefaultTestnetGoldenATXId().Hash32().Hex(),
+		GenesisID:   CalcGenesisId([]byte(DefaultTestnetGenesisExtraData()), DefaultTestnetGenesisTime()).Hex(),
+		Path:        DefaultGenesisDataDir(),
+	}
+}
+
+func DefaultTestGenesisConfig() *GenesisConfig {
+	return &GenesisConfig{
+		Accounts:    apiConfig.DefaultTestGenesisAccountConfig(),
+		GenesisTime: DefaultTestGenesisTime(),
+		ExtraData:   DefaultTestGenesisExtraData(),
+		GoldenATXID: DefaultTestGoldenATXId().Hash32().Hex(),
+		GenesisID:   CalcGenesisId([]byte(DefaultTestGenesisExtraData()), DefaultTestGenesisTime()).Hex(),
+		Path:        DefaultGenesisDataDir(),
+	}
+}
+
+func (gc *GenesisConfig) Compare(payload *GenesisConfig) error {
+	if diff := cmp.Diff(payload, gc); diff != "" {
+		fmt.Printf("config files mismatch (-want +got):\n%s", diff)
+		return errors.New("failed to match config files")
+	}
+	return nil
+}
+
+func DefaultGenesisDataDir() string {
+	return filepath.Join(defaultDataDir, DefaultGenesisConfigFileDir)
+}
+
+func DefaultGenesisTime() string {
+	return time.Now().Format(time.RFC3339)
+}
+
+func DefaultGenesisExtraData() string {
+	return "mainnet"
+}
+
+func DefaultGoldenATXId() types.ATXID {
+	return types.ATXID(CalcGoldenATX([]byte(DefaultGenesisExtraData()), DefaultGenesisTime()))
+}
+
+func DefaultTestGenesisTime() string {
+	return "2022-12-25T00:00:00+00:00"
+}
+
+func DefaultTestGenesisExtraData() string {
+	return DefaultGenesisExtraData()
+}
+
+func DefaultTestGoldenATXId() types.ATXID {
+	return types.ATXID(CalcGoldenATX([]byte(DefaultTestGenesisExtraData()), DefaultTestGenesisTime()))
+}
+
+func DefaultTestnetGenesisTime() string {
+	return DefaultTestGenesisTime()
+}
+
+func DefaultTestnetGenesisExtraData() string {
+	return "testnet"
+}
+
+func DefaultTestnetGoldenATXId() types.ATXID {
+	return types.ATXID(CalcGoldenATX([]byte(DefaultTestnetGenesisExtraData()), DefaultTestnetGenesisTime()))
+}
+
+func CalcGenesisId(genesisExtraData []byte, genesisTime string) types.Hash20 {
```

Comment:
i don't understand what this function is doing it should be

hash.Write(gentime)
hash.Write(extradata)

Reply:
this is what it does, not sure what you mean

Reply:
remove padding

Reply:
done